src/main/java/org/truffleruby/language/control/CheckIfPatternsMatchedNode.java

```diff
@@ -0,0 +1,35 @@
+/*
+ * Copyright (c) 2013, 2021 Oracle and/or its affiliates. All rights reserved. This
+ * code is released under a tri EPL/GPL/LGPL license. You can use it,
+ * redistribute it and/or modify it under the terms of the:
+ *
+ * Eclipse Public License version 2.0, or
+ * GNU General Public License version 2, or
+ * GNU Lesser General Public License version 2.1.
+ */
+package org.truffleruby.language.control;
+
+import org.truffleruby.language.RubyContextSourceNode;
+
+import com.oracle.truffle.api.frame.VirtualFrame;
+import org.truffleruby.language.RubyNode;
+
+public final class CheckIfPatternsMatchedNode extends RubyContextSourceNode {
+
+    @Child RubyNode inspected;
+
+    public CheckIfPatternsMatchedNode(RubyNode inspected) {
+        this.inspected = inspected;
+    }
+
+    @Override
+    public Object execute(VirtualFrame frame) {
+        String message = inspected.execute(frame).toString();
```

Comment:
See https://github.com/oracle/truffleruby/actions/runs/3507592176/jobs/5887990366
`org.truffleruby.language.control.CheckIfPatternsMatchedNode.execute(CheckIfPatternsMatchedNode.java:27) ; java.util.AbstractCollection.toString() ; java.lang.Object.toString()`
which means `CheckIfPatternsMatchedNode.java:27` calls toString but that can't be called in PE code.
It's best to use a ToJavaStringNode here to convert from Ruby String to Java String.

Reply:
I am not entirely sure how to use  `ToJavaStringNode` with a `RubyString`. Will `public String getJavaString()` from `TStringUtils` work?

Reply:
No, that's neverPartOfCompilation() so also not allowed in PE code.
Use `@Child  ToJavaStringNode  toJavaStringNode` + `toJavaStringNode.execute(inspected.execute(frame)`

Reply:
can i do `inspected.execute(frame).asTruffleString()` or `asTruffleStringUncached()`?

Reply:
this works:
```
        toJavaStringNode = TruffleString.ToJavaStringNode.create();
        String message = toJavaStringNode.execute(((RubyString) inspected.execute(frame)).asTruffleStringUncached());
        throw new RaiseException(getContext(), coreExceptions().noMatchingPatternError(message, this));
```

Reply:
Anything uncached is not allowed in PE code (e.g., asTruffleStringUncached).
What does `inspected.execute(frame)` return? Isn't it already a Ruby String?

Reply:
`inspected.execute(frame)` returns a `RubyString`, but `toJavaStringNode` expects a truffleString. I get a typeError regardless of a cast.

Reply:
It should be `org.truffleruby.interop.ToJavaStringNode` and not `com.oracle.truffle.api.strings.TruffleString.ToJavaStringNode`

Reply:
I have implemented the change. I hope it is correct.